feat : 카테고리 삭제

beetrbgus opened the pull request:
## 📃 관련 이슈
- #35
- #41 의 Merge가 선행 되어야 함.
## ✨ 변경 내용
- 카테고리 삭제 (비활성화 추가)

## 🔎 참고 사항 (선택)
- 추가 설명, 참고 링크 등

beetrbgus committed:
feat : 삭제 추가

Jiihyun commented on showhive-boot/admin/src/main/java/com/showhive/admin/interfaces/category/resource/CategoryResource.java:
```diff
@@ -0,0 +1,41 @@
+    }
+
+    @Override
+    @DeleteMapping("/{id}")
```
R: `categoryId`!!

Jiihyun commented on showhive-infra/infra-persistence/src/main/java/com/showhive/category/CategoryCommandRepositoryImpl.java:
```diff
@@ -0,0 +1,32 @@
+
+    @Override
+    public void deActiveCategory(Long id) {
+        Category category = queryRepository.findByIdAndIsActiveIsTrue(id)
+                .orElseThrow(() -> new CategoryException(CategoryErrorCode.CATEGORY_NOT_FOUND));
+
+        category.deactivate();
+
+        commandRepository.save(category);
```
R: 이 로직들은 레포말고 usecase쪽에 구현하면 좋을 것 같아요!
그리고 이부분도 category가 이미 영속성 컨텍스트에 있으니 더티체킹 되어서 또 save 메서드 호출 안해도 되지 않을까요??

beetrbgus replied:
더티체킹은 아까 말씀드렸던 대로 유지보수상 이점을 위한 가독성 향상이라고 생각하시면 될 것 같고- 
전자는 그게 맞을 것 같습니당

Jiihyun reviewed (commented):
굿

beetrbgus committed:
Merge branch 'develop' into feat/35-delete

beetrbgus committed:
feat : 피드백 수정

##############################
## 타입 설정                  ##
##############################

## feat: 기능 구현
## refactor: 코드 구조 개선, 변수명 변경 등 코드 리팩토링
## fix: 버그 또는 오류 해결
## style: CSS 등 스타일 관련 변경
## test: 테스트 코드 추가 또는 수정
## docs: README.md 등 문서 관련 변경
## chore: 파일 또는 디렉토리 구조 변경, 패키지 설치 등 빌드 시스템이나 외부 도구 관련 작업
## build: CI/CD 관련 변경

beetrbgus committed:
Merge remote-tracking branch 'origin/feat/35-delete' into feat/35-delete

# Conflicts:
#	showhive-boot/admin/src/main/java/com/showhive/admin/interfaces/category/resource/CategoryResource.java

beetrbgus committed:
feat : 피드백 수정

##############################
## 타입 설정                  ##
##############################

## feat: 기능 구현
## refactor: 코드 구조 개선, 변수명 변경 등 코드 리팩토링
## fix: 버그 또는 오류 해결
## style: CSS 등 스타일 관련 변경
## test: 테스트 코드 추가 또는 수정
## docs: README.md 등 문서 관련 변경
## chore: 파일 또는 디렉토리 구조 변경, 패키지 설치 등 빌드 시스템이나 외부 도구 관련 작업
## build: CI/CD 관련 변경

beetrbgus committed:
fix : 오류 수정

##############################
## 타입 설정                  ##
##############################

## feat: 기능 구현
## refactor: 코드 구조 개선, 변수명 변경 등 코드 리팩토링
## fix: 버그 또는 오류 해결
## style: CSS 등 스타일 관련 변경
## test: 테스트 코드 추가 또는 수정
## docs: README.md 등 문서 관련 변경
## chore: 파일 또는 디렉토리 구조 변경, 패키지 설치 등 빌드 시스템이나 외부 도구 관련 작업
## build: CI/CD 관련 변경
